discord_typings/_resources/_guild.py

```diff
@@ -235,6 +235,7 @@ class GuildMemberData(TypedDict):
     pending: NotRequired[bool]
     permissions: NotRequired[str]
     communication_disabled_until: NotRequired[Optional[str]]
+    source_invite_code: str
```

Comment:
As shown in the discussion you linked, this can be null, which means this should be marked as possibly None.

Additionally, due to how it is seemingly only passed in one endpoint, and because it is unofficial, I would recommend marking it as possibly missing. This also means it will likely interfere less with existing code.

```suggestion
    source_invite_code: NotRequired[Optional[str]]
```